### DISCUSSION TITLE
Some problems in using flask-socketio

### DISCUSSION THREAD

**Discussion by zhenzi0322** [miguelgrinberg/Flask-SocketIO#1604] [Q&A]
I would like to know how to change request.sid to my own id.

**Comment by miguelgrinberg** [comment]
This is not possible. Your application must create a map between the `sid` values assigned by the Socket.IO protocol to your own user ids.

**Comment by zhenzi0322** [comment]
So how do I get them to communicate with each other using my own membership form?

**Comment by miguelgrinberg** [comment]
As I said, your application must keep track of the `sid` values and associate them with the user ids that you use. Then as needed you convert from sid to user id or viceversa.

**Comment by zhenzi0322** [comment]
Will sid generate duplicate values?

**Comment by miguelgrinberg** [comment]
No.
